src/vxingest/prepbufr_to_cb/prepbufr_builder.py

```diff
@@ -0,0 +1,2007 @@
+                    else:
+                        if dir_dif < -180:
+                            dir_dif += 360
+                    value = next_higher_pressure_direction + fact * (dir_dif)
+                    if value < 0:
+                        value += 360
+                    else:
+                        if value > 360:
+                            value -= 360
+                    return value
+            else:  # if it isn't a wind_direction do this
+                next_higher_pressure_variable_value = obs_data[variable][
+                    nearest_higher_pressure_index
+                ]
+                next_lower_pressure_variable_value = obs_data[variable][
+                    nearest_lower_pressure_index
+                ]
+                if not self.is_a_number(
+                    next_higher_pressure_variable_value
+                ) or not self.is_a_number(next_lower_pressure_variable_value):
+                    return None
+                else:
+                    value = next_higher_pressure_variable_value + fact * (
+                        abs(
+                            next_higher_pressure_variable_value
+                            - next_lower_pressure_variable_value
+                        )
+                    )
+                    return value
+        except Exception as _e:
+            logging.error(
+                "PrepBufrBuilder.interpolate_level: Exception  error: %s", str(_e)
+            )
+            return None
+
+    def interpolate_data(self, raw_obs_data):
+        """fill in the mandatory levels with interpolated data using the log difference interpolation method.
+        For each pressure level in the mandatory levels, find the nearest higher and lower pressure levels
+        and interpolate the data for each variable at the mandatory level. Set the pressure level to the mandatory level.
+        Args:
+            raw_data (): this is the raw data from the prepbufr file with missing heights having been interpolated
+            using the hypsometric equation for thickness.
+        Returns: the interpolated_data
+        """
+        try:
+            interpolated_data = {}
+            for station in raw_obs_data:
+                if station not in interpolated_data:
+                    interpolated_data[station] = {}
+                for report in raw_obs_data[station]:
+                    if report not in interpolated_data[station]:
+                        interpolated_data[station][report] = {}
+                        if "data" not in interpolated_data[station][report]:
+                            interpolated_data[station][report]["data"] = {}
+                    if not isinstance(
+                        raw_obs_data[station][report]["obs_data"]["pressure"],
+                        list,
+                    ):
+                        # I cannot process this station - there is no array of pressure data
+                        del interpolated_data[station]
```

Comment:
I prefer to remove it, myself, since it is a totally invalid subset without pressure. Setting it to None just makes things harder in other parts of the code, and knowing that we have a station with no data doesn't have any value.

Reply:
Gotcha. That makes sense to me. Thanks for expanding on that. 🙂